app/controllers/application_controller.rb

```diff
@@ -7,12 +7,16 @@ def current_user
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
 
+  def require_user
+    render file: 'public/404', status: :not_found unless current_user
+  end
+
   def generate_flash(resource)
-    error_messages = String.new
+    error_messages = ''
     resource.errors.messages.each do |validation, message|
       error_messages += "#{validation.capitalize} (#{message[0]}); "
     end
-    error_messages = error_messages.delete_suffix("; ")
+    error_messages = error_messages.delete_suffix('; ')
     flash[:failure] = "ERROR: #{error_messages}"
```

Comment:
I would like to learn more about this!  Does this have to do with what you mentioned on Sunday that you figured out how to create your own flash error messages?

Reply:
I would love to show you adding our own flash messages (it actually lives in user.rb on like 24!)
The method we are currently looking at is accessing the .errors that currently exist to return the correct error messages as flash messages :)